app/domain/export/tabular/people/group_roles/mailing_list.rb

```diff
@@ -13,22 +13,15 @@ def initialize(mailing_list)
     end
 
     def as_options
-      { restrict_to_roles: role_sti_names,
-        restrict_to_group_ids: group_ids }
+      { role_restrictions: role_restrictions }
     end
 
     private
 
-    def role_sti_names
-      sti_names = []
-      group_subscriptions.each do |s|
-        sti_names += s.role_types
+    def role_restrictions
+      group_subscriptions.each_with_object({}) do |s, restrictions|
+        restrictions[s.subscriber_id] = s.role_types
```

Comment:
Das überschreibt die `role_types` der Gruppe, wenn dieselbe Gruppe mehr als einmal vorkommt, nicht?

Reply:
Ausserdem ist das sowieso falsch: `s.subscriber_id` ist nicht zwingend die Gruppe in der die `role_types` vorkommen. Z.B. kann `subscriber_id` die ID des Dachverbands sein, aber die `role_types` können Rollen-Typen in Untergruppen des Dachverbands sein.
Bei den MailingLists werden die eigentlichen GruppenID/RollenTyp-Paare glaube ich nie wirklich berechnet, aber für die Anzeige werden sie mit diesem Code gruppiert: https://github.com/hitobito/hitobito/blob/master/app/models/subscription.rb#L78..L90
Dieser Code kann aber leider nicht 1:1 hier wiederverwendet werden, weil `Role::TypeList` nicht Gruppen-IDs sondern Gruppennamen ausspuckt. Daher müssten wir diese Logik hier nochmals nachprogrammieren.